src/main.rs

```diff
@@ -41,82 +51,67 @@ fn main() {
         .arg(
             Arg::with_name("net")
                 .long("net")
-                .help("Network parameters \"tap=<if_name>,ip=<ip_addr>,mask=<net_mask>,mac=<mac_addr>\"")
-                .takes_value(true),
-        )
-        .arg(
-            Arg::with_name("cpus")
-                .long("cpus")
-                .help("Number of virtual CPUs")
-                .takes_value(true),
-        )
-        .arg(
-            Arg::with_name("memory")
-                .long("memory")
-                .help("Amount of RAM (in MB)")
+                .help(
+                    "Network parameters \"tap=<if_name>,\
+                     ip=<ip_addr>,mask=<net_mask>,mac=<mac_addr>\"",
+                )
                 .takes_value(true),
         )
         .arg(
             Arg::with_name("rng")
                 .long("rng")
                 .help("Path to entropy source")
-                .default_value("/dev/urandom"),
+                .default_value(config::DEFAULT_RNG_SOURCE),
         )
         .get_matches();
 
-    let kernel_arg = cmd_arguments
+    // These .unwrap()s cannot fail as there is a default value defined
+    let cpus = cmd_arguments.value_of("cpus").unwrap();
+    let memory = cmd_arguments.value_of("memory").unwrap();
+
+    let kernel = cmd_arguments
         .value_of("kernel")
-        .map(PathBuf::from)
         .expect("Missing argument: kernel");
-    let kernel_path = kernel_arg.as_path();
 
-    let disk_paths: Vec<PathBuf> = cmd_arguments
+    let cmdline = cmd_arguments.value_of("cmdline");
+
+    let disks: Vec<&str> = cmd_arguments
         .values_of("disk")
-        .expect("Missing arguments on disk")
-        .map(PathBuf::from)
+        .expect("Missing argument: disk. Provide at least one")
         .collect();
 
-    let cmdline = cmd_arguments
-        .value_of("cmdline")
-        .map(std::string::ToString::to_string)
-        .unwrap_or_else(String::new);
+    let net = cmd_arguments.value_of("net");
 
-    let net_params = cmd_arguments
-        .value_of("net")
-        .map(std::string::ToString::to_string);
+    let rng = cmd_arguments.value_of("rng");
```

Comment:
You might as well unwrap this too as it is currently always set.

Reply:
Done